src/include/miopen/solver.hpp

```diff
@@ -246,7 +246,11 @@ struct PerformanceConfigConvAsm3x3U : Serializable<PerformanceConfigConvAsm3x3U>
     bool IsValidValue() const;
     bool SetNextValue(const ConvolutionContext& config);
     bool IsValid(const ConvolutionContext& config) const;
-    bool operator==(const PerformanceConfigConvAsm3x3U& other) const;
+    inline bool operator==(const PerformanceConfigConvAsm3x3U& other) const
```

Comment:
Because of the changes in generic_search, inline can't be used when the operator is defined in the cpp file. 
The function definition could stay in the cpp file, but it would be declared as a standard function.
To use inline we would need to move the function to the header.
Per Jehandad's comment I moved the function to the header to use inline to avoid function call cost.

Reply:
It shouldn't have noticeable performance difference in any meaningful case (unless 0 configs were valid). For a couple calls to this operator on average you would have a compilation and kernel launch. Couple int comparison would have no noticeable effect even with high precision clock. At the same time the header is already massive. And these calls would probably be virtualized later anyway.
Also modern day linkers actually can inline stuff from other cpp files sometimes.
Overall I would like you to revert the inlining of comparison operators for perf config.